### PR TITLE
remove tika and refactor FileUtil and FileHelper

### PR DESCRIPTION
**Pull request by Siedlerchr**
```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by Siedlerchr** [comment]
failing unit test is not related

File: src/main/java/org/jabref/gui/desktop/JabRefDesktop.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ private static void openDoi(String doi) throws IOException {
 
     public static void openCustomDoi(String link, PreferencesService preferences, DialogService dialogService) {
             DOI.parse(link)
-                        .map(identifier -> (DOI) identifier)
+                        .map(identifier -> identifier)
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/desktop/JabRefDesktop.java:134]: ?

File: src/main/java/org/jabref/logic/net/URLDownload.java [hunk 2]
```diff
@@ -204,11 +204,11 @@ public String getMimeType() {
...
 
-       // Set a custom Apache Client Builder to be able to allow circular redirects, otherwise downloads from springer might not work
+        // Set a custom Apache Client Builder to be able to allow circular redirects, otherwise downloads from springer might not work
         Unirest.config().httpClient(new ApacheClient.Builder()
-                                    .withRequestConfig((c, r) -> RequestConfig.custom()
-                                                       .setCircularRedirectsAllowed(true)
-                                                       .build()));
+                                                              .withRequestConfig((c, r) -> RequestConfig.custom()
+                                                                                                        .setCircularRedirectsAllowed(true)
+                                                                                                        .build()));
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/net/URLDownload.java:211]: indentation

File: src/main/java/org/jabref/logic/util/FileUtil.java [hunk 3]
```diff
@@ -308,26 +326,96 @@ public static String createDirNameFromPattern(BibDatabase database, BibEntry ent
      * @param rootDirectory the rootDirectory that will be searched
      * @return the path to the first file that matches the defined conditions
      */
-    public static Optional<Path> find(String filename, Path rootDirectory) {
+    public static Optional<Path> findSingleOld(String filename, Path rootDirectory) {
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/util/FileUtil.java:329]: So if this is deprecated, add `@deprecated`?

File: src/main/java/org/jabref/model/entry/LinkedFile.java [hunk 4]
```diff
@@ -17,14 +17,16 @@
...
+import org.jabref.logic.util.FileUtil;
 import org.jabref.model.database.BibDatabaseContext;
-import org.jabref.model.util.FileHelper;
 import org.jabref.preferences.FilePreferences;
 
 /**
  * Represents the link to an external file (e.g. associated PDF file).
  * This class is {@link Serializable} which is needed for drag and drop in gui
  */
+@AllowedToUseLogic("Uses FileUtil from logic")
```
Comment (calixtus) [inline src/main/java/org/jabref/model/entry/LinkedFile.java:29]: Can FileUtil be moved to the model domain?

**Review by calixtus — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/model/entry/LinkedFile.java (hunk 4 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/model/entry/LinkedFile.java:29]: nope, tried this before not possible

File: src/main/java/org/jabref/logic/util/FileUtil.java (hunk 3 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/util/FileUtil.java:329]: I don't think its deprecated, will give it another name

File: src/main/java/org/jabref/gui/desktop/JabRefDesktop.java (hunk 1 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/desktop/JabRefDesktop.java:134]: seems to be have done automatically by eclipe

File: src/main/java/org/jabref/model/entry/LinkedFile.java (hunk 4 above)
Comment (koppor) [reply src/main/java/org/jabref/model/entry/LinkedFile.java:29]: And handling files is really a "logic" thing, not a model thing. We can talk about refactoring the "model" package as some things are do not sound like "model"

image

File: src/main/java/org/jabref/gui/DragAndDropHelper.java [hunk 5]
```diff
@@ -8,7 +8,7 @@
 
 import javafx.scene.input.Dragboard;
 
-import org.jabref.logic.util.io.FileUtil;
+import org.jabref.logic.util.FileUtil;
```
Comment (koppor) [inline src/main/java/org/jabref/gui/DragAndDropHelper.java:11]: Why move it a level up? In the io package, there are many other libraries - and I like the grouping

image

File: src/test/resources/org/jabref/logic/util/pdffile.pdf
Comment (koppor) [inline src/test/resources/org/jabref/logic/util/pdffile.pdf:1]: I see no deletion of `pdffile.pdf`. Do we really need to keep it twice?

File: src/test/java/org/jabref/logic/util/FileUtilTest.java [hunk 6]
```diff
@@ -351,15 +354,90 @@ void testIsNotBibFile() throws IOException {
...
     @Test
     void testFindinPath() {
-        Optional<Path> resultPath1 = FileUtil.find("existingTestFile.txt", rootDir);
+        Optional<Path> resultPath1 = FileUtil.findSingleFileRecursively("existingTestFile.txt", rootDir);
         assertEquals(resultPath1.get().toString(), existingTestFile.toString());
     }
 
     @Test
     void testFindInListOfPath() {
+        // due to the added workaround for old JabRef behavior as both path starts with the same name they are considered equal
```
Comment (koppor) [inline src/test/java/org/jabref/logic/util/FileUtilTest.java:363]: However, the behavior changed. Not listed twice, which is good.

**Review by koppor — COMMENTED** [review]
Two small comments

File: src/test/java/org/jabref/logic/util/FileUtilTest.java (hunk 6 above)
Comment (koppor) [reply src/test/java/org/jabref/logic/util/FileUtilTest.java:363]: (no action needs to be taken)

**Pull request merged**